assets/blocks/shared.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * WordPress dependencies
+ */
+import { unregisterBlockType } from '@wordpress/blocks';
+import { subscribe, select } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import registerSenseiBlocks from './register-sensei-blocks';
+import ContactTeacherBlock from './contact-teacher-block';
+
+const BLOCKS_PER_POST_TYPE = {
+	'sensei-lms/button-contact-teacher': [ 'course', 'lesson' ],
+};
+
+registerSenseiBlocks( [ ContactTeacherBlock ] );
+
+let postType = null;
+
+subscribe( () => {
+	if ( postType ) {
+		return;
+	}
+
+	postType = select( 'core/editor' ).getCurrentPostType();
+
+	if ( ! postType ) {
+		return;
+	}
+
+	// Unregister blocks that should not appear in certain post types.
+	Object.entries( BLOCKS_PER_POST_TYPE ).forEach(
```

Comment:
It unregisters the shared block based on the current post type. Now that we have blocks only in the `lesson` and `course`, it doesn't make any difference, but when we have shared blocks, including other post types, it'll be necessary.

Basically, we'll include the shared blocks file for all post types that contain shared blocks, but it doesn't mean that we want all blocks appearing in any post type.

Reply:
Maybe we should just ditch the `shared` assets, and include shared blocks both in the `single-lesson` and `single-course` bundles. It means some code duplication, but removes this complexity, and it doesn't really make a difference in performance. Also, library code that's shared, like helpers, are already compiled into all of the bundles.

Reply:
I don't like the idea to duplicate the code in different assets if we have an alternative to avoid that. When we create more shared blocks and add them to more contexts, this can accumulate a lot of code duplication in different assets.

> Also, library code that's shared, like helpers, are already compiled into all of the bundles.

About this, I didn't explore too much, but I tried to see if there was a configuration in the `DependencyExtractionWebpackPlugin` to extract these local helpers too. But it seems there is not and it would be a little complicated. But when it grows too much, we can also explore a solution for that.